message_definitions/v1.0/ardupilotmega.xml

```diff
@@ -93,37 +93,37 @@
       <entry value="42004" name="MAV_CMD_FIXED_MAG_CAL">
         <description>Magnetometer calibration based on fixed position
         in earth field given by inclination, declination and intensity.</description>
-        <param index="1">MagDeclinationDegrees.</param>
-        <param index="2">MagInclinationDegrees.</param>
-        <param index="3">MagIntensityMilliGauss.</param>
-        <param index="4">YawDegrees.</param>
+        <param index="1" label="Declination" units="deg">MagDeclinationDegrees.</param>
+        <param index="2" label="Inclination" units="deg">MagInclinationDegrees.</param>
+        <param index="3" label="Intensity" units="mgauss">MagIntensityMilliGauss.</param>
+        <param index="4" label="Yaw" units="deg">YawDegrees.</param>
         <param index="5">Empty.</param>
         <param index="6">Empty.</param>
         <param index="7">Empty.</param>
       </entry>
       <entry value="42005" name="MAV_CMD_FIXED_MAG_CAL_FIELD">
-        <description>Magnetometer calibration based on fixed expected field values in milliGauss.</description>
-        <param index="1">FieldX.</param>
-        <param index="2">FieldY.</param>
-        <param index="3">FieldZ.</param>
+        <description>Magnetometer calibration based on fixed expected field values.</description>
+        <param index="1" label="Field X" units="mgauss">FieldX.</param>
+        <param index="2" label="Field Y" units="mgauss">FieldY.</param>
+        <param index="3" label="Field Z" units="mgauss">FieldZ.</param>
         <param index="4">Empty.</param>
         <param index="5">Empty.</param>
         <param index="6">Empty.</param>
         <param index="7">Empty.</param>
       </entry>
       <entry value="42424" name="MAV_CMD_DO_START_MAG_CAL">
         <description>Initiate a magnetometer calibration.</description>
-        <param index="1">uint8_t bitmask of magnetometers (0 means all).</param>
-        <param index="2">Automatically retry on failure (0=no retry, 1=retry).</param>
-        <param index="3">Save without user input (0=require input, 1=autosave).</param>
-        <param index="4">Delay (seconds).</param>
-        <param index="5">Autoreboot (0=user reboot, 1=autoreboot).</param>
+        <param index="1" label="Bitmask" minValue="0" maxValue="255" increment="1">Bitmask of magnetometers to calibrate (0 means all).</param>
```

Comment:
Label could do with some help.

Reply:
.... also, don't we have a display intent mechanism for this sort of thing?  `display="bitmask"` in fact.

It would be nice to be able to capture the "value of 0 means all" thing in this XML too.

Reply:
Yes, "Magnetometer Bitmask".

> It would be nice to be able to capture the "value of 0 means all" thing in this XML too.

The use of 0 for all is crap - doesn't "0" mean none? and 255 by definition mean all?

Reply:
@peterbarker 
- As above, any way we can change that "0 means all" thing to the more sane "255 means all". It is idiotic.
- `display="bitmask"` only applies to message fields. @DonLakeFlyer - would this be useful for params too?

@amilcarlucas Either way, the label should change to `label="Magnetometer Bitmask"` or `label="Magnetometer Flags"`

Reply:
> The use of 0 for all is crap - doesn't "0" mean none? and 255 by definition mean all?

Yes, but at the same time the command to start calibrating none makes no sense either.

The more specific reason we don't accept 255 is that if you indicate we want to calibrate a compass, but it isn't healthy, or isn't used for yaw we nack the command, because you requested calibrating something that couldn't be calibrated. If you chose to specify what should be calibrated with the bitmask that's taken as a strict requirement.

Reply:
@WickedShell Ah, so what I think you are saying is that 0 does **not** mean "all" it means "*attempt* to calibrate all magnetometers that should be calibratable - ie not those that are not used for yaw or not healthy." This contrasts with 255, which would mean "attempt to calibrate all magnetometers, whether enabled or unhealthy or whatever". Is that correct?

That makes sense, but should be better clarified.

Reply:
Correct. (And I agree that the description could have been better). From a description perspective "0 means attempt to calibrate all sensors, and ignore any sensors for which a calibration could not be started." You could maybe tag on a bit about "Reasons for a calibration to not start can include, but is not limited to, a sensor being unhealthy, or disabled.

Reply:
@WickedShell Thanks. Clear now!  I'd quite like to more clearly differentiate from 255. How about:

> Bitmask of magnetometers to calibrate.  Use 0 to calibrate all sensors that can be started (sensors may not start if disabled, unhealthy, etc.). Note that using 0 differs from 255, which will NACK if calibration does not complete for all sensors.

Reply:
It's not that 255 is special, it's whenever you flag any specific one it's required.

Reply:
@WickedShell I understand that, but I was hoping to clarify the difference between two methods that both "start everything".

Perhaps 

> Bitmask of magnetometers to calibrate. Use 0 to calibrate all sensors that can be started (sensors may not start if disabled, unhealthy, etc.). The command will NACK if calibration does not succeed for a sensor specified by the bitmask.

Reply:
The command NACK's if it can't start a calibration. Calibration failure is reported in a different message entirely.

Reply:
Grrr. 

> Bitmask of magnetometers to calibrate. Use 0 to calibrate all sensors that can be started (sensors may not start if disabled, unhealthy, etc.). The command will NACK if calibration does not start for a sensor explicitly specified by the bitmask.